linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -1572,22 +1572,29 @@ private object AnalyzerRun {
 
   private class WorkTracker(implicit ec: ExecutionContext) {
     private val pending = new AtomicInteger(0)
+    private val failures = new AtomicReference[List[Throwable]](Nil)
     @volatile private var _allowComplete = false
     private val promise = Promise[Unit]()
 
     def track(fut: Future[Unit]): Unit = {
       pending.incrementAndGet()
 
-      fut.onComplete {
-        case Success(_) =>
-          if (pending.decrementAndGet() == 0)
-            tryComplete()
-
-        case Failure(t) =>
-          promise.tryFailure(t)
+      fut.onComplete { result =>
+        result match {
```

Comment:
```suggestion
        result.failed.foreach(addFailure(_))
```
?

Reply:
I think I disagree about this one. I like `.failed` when I *expect* the failure, such as in test cases. But when I expect success, IMO using `.failed` puts too much emphasis on the "sad path" rather than the "happy path".

Does that make sense?

Reply:
I don't necessarily agree with the sentiment, but I'm fine with the way it currently is.

The current code has the clear advantage of being very explicit that both cases have been considered.